sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -28,6 +28,22 @@ import org.apache.spark.unsafe.types.{CalendarInterval, UTF8String}
 
 object IntervalUtils {
 
+  object IntervalUnit extends Enumeration {
+    type IntervalUnit = Value
+
+    val Nanosecond = Value(0, "nanosecond")
+    val Microsecond = Value(1, "microsecond")
+    val Millisecond = Value(2, "millisecond")
+    val Second = Value(3, "second")
+    val Minute = Value(4, "minute")
+    val Hour = Value(5, "hour")
+    val Day = Value(6, "day")
+    val Week = Value(7, "week")
+    val Month = Value(8, "month")
+    val Year = Value(9, "year")
```

Comment:
Could you use all capitals like `YEAR` please, @MaxGekk ?

Reply:
This contradicts to the Scala coding style guide: https://github.com/databricks/scala-style-guide#naming-convention `Enums should be PascalCase.`

Reply:
I know already, but it's not in Apache Spark. That's the reason I ask like this. https://github.com/apache/spark/pull/26455#issuecomment-552143074 .